### PR TITLE
Fix TensorBoard error with OBB

### LINKED ISSUE #23265: TensorBoard graph visualization failure
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug
```
Ultralytics 8.4.3 🚀 Python-3.11.14 torch-2.9.1+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=1, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=dota8.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=10, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=320, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo26n-obb.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train4, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False, project=None, rect=False, resume=False, retina_masks=False, rle=1.0, save=True, save_conf=False, save_crop=False, save_dir=/yolo/runs/obb/train4, save_frames=False, save_json=False, save_period=-1, save_txt=False, scale=0.5, seed=0, shear=0.0, show=False, show_boxes=True, show_conf=True, show_labels=True, simplify=True, single_cls=False, source=None, split=val, stream_buffer=False, task=obb, time=None, tracker=botsort.yaml, translate=0.1, val=True, verbose=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=3.0, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None
TensorBoard: Start with 'tensorboard --logdir /yolo/runs/obb/train4', view at http://localhost:6006/

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      6640  ultralytics.nn.modules.block.C3k2            [32, 64, 1, False, 0.25]
  3                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
  4                  -1  1     26080  ultralytics.nn.modules.block.C3k2            [64, 128, 1, False, 0.25]
  5                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
  6                  -1  1     87040  ultralytics.nn.modules.block.C3k2            [128, 128, 1, True]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  1    346112  ultralytics.nn.modules.block.C3k2            [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5, 3, True]
 10                  -1  1    249728  ultralytics.nn.modules.block.C2PSA           [256, 256, 1]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  1    119808  ultralytics.nn.modules.block.C3k2            [384, 128, 1, True]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  1     34304  ultralytics.nn.modules.block.C3k2            [256, 64, 1, True]
 17                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  1     95232  ultralytics.nn.modules.block.C3k2            [192, 128, 1, True]
 20                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  1    463104  ultralytics.nn.modules.block.C3k2            [384, 256, 1, True, 0.5, True]
 23        [16, 19, 22]  1    390360  ultralytics.nn.modules.head.OBB26            [15, 1, 1, True, [64, 128, 256]]
YOLO26n-obb summary: 291 layers, 2,652,984 parameters, 2,652,984 gradients, 6.3 GFLOPs

Transferred 792/792 items from pretrained weights
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2151.0±567.7 MB/s, size: 227.3 KB)
train: Scanning /yolo/datasets/dota8/labels/train... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 712.4it/s 0.0s
train: New cache created: /yolo/datasets/dota8/labels/train.cache
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1418.0±968.9 MB/s, size: 97.3 KB)
val: Scanning /yolo/datasets/dota8/labels/val... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 1.2Kit/s 0.0s
val: New cache created: /yolo/datasets/dota8/labels/val.cache
Plotting labels to /yolo/runs/obb/train4/labels.jpg...
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: MuSGD(lr=0.000526, momentum=0.9) with parameter groups 0 weight(decay=0.0), 0 weight(decay=0.0005), 24 bias(decay=0.0)
WARNING ⚠️ TensorBoard: TensorBoard graph visualization failure: Tracer cannot infer type of (tensor([[[ 5.1451e+01,  5.8704e+01,  8.8570e+01,  ...,  5.0365e-05,  7.0000e+00,  4.3944e-03],
         [ 7.6229e+01,  2.0657e+01,  1.1588e+02,  ...,  4.1951e-05,  7.0000e+00, -1.5607e-03],
         [ 5.1450e+01,  2.0277e+01,  8.3306e+01,  ...,  4.0920e-05,  7.0000e+00,  3.1936e-03],
         ...,
          [ 1.7078,  0.8693,  1.5679,  ..., -0.2693,  0.6455, -0.2170],
          [ 1.2936,  0.4024,  1.4049,  ..., -0.1055,  1.1518,  0.2598],
          [ 0.5727,  0.5314,  0.4425,  ...,  0.0149, -0.0866, -0.0448]]]], device='cuda:0')], 'angle': tensor([[[-0.0145, -0.0043,  0.0059,  ..., -0.0026,  0.0023, -0.0076]]], device='cuda:0', grad_fn=<CatBackward0>)}})
:Dictionary inputs to traced functions must have consistent type. Found Tensor and List[Tensor] -> Only Tensors created explicitly by the user (graph leaves) support the deepcopy protocol at the moment.  If you were attempting to deepcopy a module, this may be because of a torch.nn.utils.weight_norm usage, see https://github.com/pytorch/pytorch/pull/103001
Image sizes 320 train, 320 val
Using 8 dataloader workers
Logging results to /yolo/runs/obb/train4
Starting training for 10 epochs...
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss angle_loss  Instances       Size
       1/10      0.17G      1.524      3.222    0.04274    0.04499         52        320: 100% ━━━━━━━━━━━━ 4/4 2.1s/it 8.2s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.3it/s 3.0s
                   all          4          8      0.982      0.954      0.995      0.868
```
### Environment
```
Ultralytics 8.4.3 🚀 Python-3.11.14 torch-2.9.1+cu128 CUDA:0 (Tesla T4, 14913MiB)
Setup complete ✅ (80 CPUs, 62.3 GB RAM, 1084.2/3665.0 GB disk)

OS                     Linux-6.8.0-86-generic-x86_64-with-glibc2.35
Environment            Linux
Python                 3.11.14
Install                pip
Path                   /home/deeplearning/miniconda3/envs/yolo26/lib/python3.11/site-packages/ultralytics
RAM                    62.33 GB
Disk                   1084.2/3665.0 GB
CPU                    Intel Xeon Silver 4316 2.30GHz
CPU count              80
GPU                    Tesla T4, 14913MiB
GPU count              4
CUDA                   12.8

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 12.1.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.17.0>=1.4.1
torch                  ✅ 2.9.1>=1.8.0
torch                  ✅ 2.9.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.24.1>=0.9.0
psutil                 ✅ 7.2.1>=5.8.0
polars                 ✅ 1.37.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18
```
### Minimal Reproducible Example

```
yolo obb train model=yolo26n-obb.pt data=dota8.yaml epochs=10 imgsz=320 batch=1
```

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by Y-T-G**
Closes #23265 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds `smart_inference_mode()` to the TensorBoard graph-logging callback to make graph export safer and more efficient 🚀📈

### 📊 Key Changes
- Imported `smart_inference_mode` from `ultralytics.utils.torch_utils`
- Decorated `_log_tensorboard_graph(trainer)` with `@smart_inference_mode()` so it runs in inference context (no gradients) 🧠⚡
- No changes to scalar logging behavior; only the model graph logging path is affected 📊

### 🎯 Purpose & Impact
- Reduces overhead when logging the model graph to TensorBoard by avoiding gradient tracking, which can improve speed and memory usage 🏎️💾
- Helps prevent accidental autograd-related side effects during graph logging, making training callbacks more robust 🛡️
- Users should see the same TensorBoard outputs, but with fewer chances of slowdowns or memory spikes when graph logging is enabled ✅📉

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message—an Ultralytics engineer will assist with review shortly. 🧩✨ To ensure a seamless integration of your work, please review the following checklist:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

This diff looks clean and targeted. Detaching `x["angle"]` before storing it on the module avoids retaining an autograd-connected tensor on `self` during OBB inference, which can prevent TensorBoard/logging graph-retention issues while keeping the actual inference output unchanged (still concatenating the original `x["angle"]`). No issues found in the changed lines.

**Review by lmycross — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged! Huge thanks @Y-T-G (and @lmycross for the teamwork) for making TensorBoard graph logging safer and leaner by wrapping `_log_tensorboard_graph()` with `@smart_inference_mode()`.

As **Leonardo da Vinci** said, *“Simplicity is the ultimate sophistication.”* This PR is a great example of that principle in action: no changes to user-facing scalar logging, just a clean, robust inference-context guard that helps avoid autograd side effects and reduces memory/overhead when exporting graphs—making training callbacks more reliable for everyone.

Appreciate you helping keep Ultralytics and YOLO11 workflows fast, stable, and polished.
